File: app/src/main/java/com/chopecard/domain/usecases/GetProductDetailUseCase.java
package com.chopecard.domain.usecases;

import com.chopecard.data.repository.ProductRepository;
import com.chopecard.domain.models.Product;

class GetProductDetailUseCase(private val productRepository: ProductRepository) {
    suspend operator fun invoke(productId: Int): Product {
        return productRepository.getProduct(productId);
    }
}

File: app/src/main/java/com/chopecard/domain/usecases/GetProductDetailUseCase.kt
// GetProductDetailUseCase.kt
package com.chopecard.domain.usecases

import com.chopecard.data.model.Card
import com.chopecard.data.repository.CardRepository

class GetProductDetailUseCase(private val cardRepository: CardRepository) {
    suspend fun execute(cardName: String): List<Card> {
        if (cardName.isEmpty()) {
            throw IllegalArgumentException("Card name cannot be empty")
        }
        return cardRepository.getYugiohCardByName(cardName)
    }
}
